### PR TITLE
Chapter9

### PR DESCRIPTION
**Pull request by AndreeaSburlea**
## New features
-prototype cell

## Changes in code
-no changes
 
## Screenshot
ch9-dark
ch9-light

### CODE REVIEW

File: FoodPin/FoodPin/RestaurantTableViewCell.swift [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+//  Created by Andreea Sburlea on 4/14/22.
+//
+
+import UIKit
+
+class RestaurantTableViewCell: UITableViewCell {
+    @IBOutlet var nameLabel: UILabel!
+    @IBOutlet var locationLabel: UILabel!
+    @IBOutlet var typeLabel: UILabel!
+    @IBOutlet var thumbnailImageView: UIImageView!
```
Comment (ralumarusca) [inline FoodPin/FoodPin/RestaurantTableViewCell.swift]: IBOutlets should be private and you should create a method to be able to modify them, something like "configureCell" which you can call from the viewController

File: FoodPin/FoodPin/RestaurantTableViewCell.swift [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+    override func awakeFromNib() {
+        super.awakeFromNib()
+        // Initialization code
+    }
+
+    override func setSelected(_ selected: Bool, animated: Bool) {
+        super.setSelected(selected, animated: animated)
+        // Configure the view for the selected state
+    }
```
Comment (ralumarusca) [inline FoodPin/FoodPin/RestaurantTableViewCell.swift]: Unused methods can be removed

File: FoodPin/FoodPin/RestaurantTableViewController.swift [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+            cellProvider: { tableView, indexPath, restaurantName in
+                guard let cell = tableView.dequeueReusableCell(withIdentifier: cellIdentifier, for: indexPath) as? RestaurantTableViewCell
+                else {
+                    return UITableViewCell()
+                }
+                cell.nameLabel.text = restaurantName
+                cell.thumbnailImageView.image = UIImage(named: self.restaurantImages[indexPath.row])
+                cell.locationLabel.text = self.restaurantLocations[indexPath.row]
+                cell.typeLabel.text = self.restaurantTypes[indexPath.row]
+        return cell
```
Comment (ralumarusca) [inline FoodPin/FoodPin/RestaurantTableViewController.swift]: Align this line with the previous rows

File: FoodPin/FoodPin/RestaurantTableViewController.swift [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    var restaurantImages = ["cafedeadend", "homei", "teakha", "cafeloisl", "petiteoyster", "forkee", "posatelier", "bourkestreetbakery", "haigh", "palomino", "upstate", "traif", "graham", "waffleandwolf", "fiveleaves", "cafelore", "confessional", "barrafina", "donostia", "royaloak", "cask"]
+    var restaurantLocations = ["Hong Kong", "Hong Kong", "Hong Kong", "Hong Kong", "Hong Kong", "Hong Kong", "Hong Kong", "Sydney", "Sydney", "Sydney", "New York", "New York", "New York", "New York", "New York", "New York", "New York", "London", "London", "London", "London"]
+    var restaurantTypes = ["Coffee & Tea Shop", "Cafe", "Tea House", "Austrian / Causual Drink", "French", "Bakery", "Bakery", "Chocolate", "Cafe", "American / Seafood", "American", "American", "Breakfast & Brunch", "Coffee & Tea", "Coffee & Tea", "Latin American", "Spanish", "Spanish", "Spanish", "British", "Thai"]
+    enum Section {
+        case all
+    }
+    lazy var dataSource = configureDataSource()
+    func configureDataSource() -> UITableViewDiffableDataSource<Section, String> {
```
Comment (ralumarusca) [inline FoodPin/FoodPin/RestaurantTableViewController.swift:22]: Ideally there should be an empty line before and after each method to be more readable

File: FoodPin/FoodPin/RestaurantTableViewController.swift [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    var restaurantImages = ["cafedeadend", "homei", "teakha", "cafeloisl", "petiteoyster", "forkee", "posatelier", "bourkestreetbakery", "haigh", "palomino", "upstate", "traif", "graham", "waffleandwolf", "fiveleaves", "cafelore", "confessional", "barrafina", "donostia", "royaloak", "cask"]
+    var restaurantLocations = ["Hong Kong", "Hong Kong", "Hong Kong", "Hong Kong", "Hong Kong", "Hong Kong", "Hong Kong", "Sydney", "Sydney", "Sydney", "New York", "New York", "New York", "New York", "New York", "New York", "New York", "London", "London", "London", "London"]
+    var restaurantTypes = ["Coffee & Tea Shop", "Cafe", "Tea House", "Austrian / Causual Drink", "French", "Bakery", "Bakery", "Chocolate", "Cafe", "American / Seafood", "American", "American", "Breakfast & Brunch", "Coffee & Tea", "Coffee & Tea", "Latin American", "Spanish", "Spanish", "Spanish", "British", "Thai"]
+    enum Section {
+        case all
+    }
```
Comment (ralumarusca) [inline FoodPin/FoodPin/RestaurantTableViewController.swift:14]: This enum can be moved at the top of the class, before the properties, so everything is grouped

**Review by ralumarusca — CHANGES_REQUESTED** [review]

**Comment by czegreanu** [comment]
please rebase and fix-up the commits

File: FoodPin/FoodPin/RestaurantTableViewController.swift [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+//  Created by Andreea Sburlea on 4/14/22.
+//
+
+import UIKit
+
+class RestaurantTableViewController: UITableViewController {
+    // swiftlint:disable line_length
+    enum Section {
+        case all
+    }
```
Comment (paulpop10) [inline FoodPin/FoodPin/RestaurantTableViewController.swift:14]: There should be an empty space after a closed brace

**Review by paulpop10 — APPROVED** [review]

**Review by ralumarusca — APPROVED** [review]

**Pull request merged**